haskell/private/ghci_repl_wrapper.sh

```diff
@@ -52,13 +52,17 @@ cd "$BUILD_WORKSPACE_DIRECTORY"
 # location of exec root reliably and then prefix locations of various
 # components, such as shared libraries with that exec root.
 
-RULES_HASKELL_EXEC_ROOT=$(dirname $(readlink ${BUILD_WORKSPACE_DIRECTORY}/bazel-out))
-TOOL_LOCATION="$RULES_HASKELL_EXEC_ROOT/{TOOL}"
+RULES_HASKELL_EXEC_ROOT=$(dirname "$(readlink "${BUILD_WORKSPACE_DIRECTORY}"/bazel-out)")
+TOOL_LOCATION="$RULES_HASKELL_EXEC_ROOT/%{TOOL}"
 # Setting -pgm* flags explicitly has the unfortunate side effect
 # of resetting any program flags in the GHC settings file. So we
 # restore them here. See
 # https://ghc.haskell.org/trac/ghc/ticket/7929.
-PGM_ARGS="-pgma {CC} -pgmc {CC} -pgml {CC} -pgmP {CC} -optc-fno-stack-protector -optP-E -optP-undef -optP-traditional"
+PGM_ARGS="-pgma %{CC} -pgmc %{CC} -pgml %{CC} -pgmP %{CC} -optc-fno-stack-protector -optP-E -optP-undef -optP-traditional"
 
-{ENV}
-"$TOOL_LOCATION" $PGM_ARGS {ARGS} "$@"
+# shellcheck disable=SC1083
+%{ENV}
+
+# Disabling 2086, because we indeed want word splitting on PGM_ARGS
+# shellcheck disable=SC1083 disable=SC2086
+"$TOOL_LOCATION" $PGM_ARGS %{ARGS} "$@"
```

Comment:
Heads-up: There will be a merge conflict with https://github.com/tweag/rules_haskell/pull/1263/files#diff-40157ed5196395bde07afb099668abf3R59.